### ISSUE TITLE
Jit compile methods with watched fields by default on Power and S390

### ISSUE THREAD

**Issue by AlenBadel** [eclipse/openj9#6824]
Currently on Power and Z, to enable JIT compilation on methods that contains watched fields -XX:JITInlineWatches must be passed in as an option.

This should happen by default, without the extra option.

See:
https://github.com/eclipse/openj9/blob/327186078fc64f3ab12d56a18fc9524833fd4cd7/runtime/vm/jvminit.c#L2977

The usage of -XX:JITInlineWatches likely needs to be changed in the docs as well.

**Comment by AlenBadel** [comment]
Looking into making the change.

FYI @gacholio

**Comment by gita-omr** [comment]
I think this should be in 0.16.

**Comment by dchopra001** [comment]
The release notes from 0.14 imply that this option is enabled by default on Z and X:
https://eclipse.github.io/openj9-docs/version0.15/#performance-improvements-for-jvmti-watched-fields
https://eclipse.github.io/openj9-docs/xxjitinlinewatches/

Looks like we forgot to actually enable it 😅.

**Comment by gita-omr** [comment]
We will need to modify the tests not to use it...

**Comment by dchopra001** [comment]
Based on how we consume the arguments here, the tests should still work with JIT support enabled even with the option specified. 
https://github.com/eclipse/openj9/blob/327186078fc64f3ab12d56a18fc9524833fd4cd7/runtime/vm/jvminit.c#L3189-L3195 

Considering that support still isn't available on ARM, do you think it'd be better to leave the option in the tests right now? That way when support is implemented on ARM, they will be able to run the tests directly without having to export extra options on the command line (which can be problematic if they only want to enable field watch support for specific tests). 

This would mean that for now we can just use the same interim solution that X is using (i.e expand the check in https://github.com/eclipse/openj9/issues/6824#issue-484168252 to check for Power and Z as well).

**Comment by gita-omr** [comment]
I think we first need to decide whether the default in  https://github.com/eclipse/openj9/blob/327186078fc64f3ab12d56a18fc9524833fd4cd7/runtime/vm/jvminit.c#L2977
should be the same for all platforms (including say ARM where it's not supported yet) eg. removing the #if above or not. I thought there was some other check somewhere that would prevent JIT compilations for platforms that don't support the feature. But I don't recall where it was...

**Comment by gacholio** [comment]
My interest here is in getting rid of the old (and not entirely working) code from the VM, so until all platforms support the watch callbacks, I think this will have to wait.

While we continue to have the option of disabling the feature, the old code paths in the JIT need to remain.

It's possible we could disallow the disable feature on certain architectures, but that doesn't seem very sensible.

**Comment by gita-omr** [comment]
My main concern is for the default to match (1) the documentation (2) the blog that we are about to post

The example in the blog makes JIT inlining detectable since the test case becomes much faster.

**Comment by gita-omr** [comment]
I am ok with leaving the option in the test case since the effect of it is not detectable there.

**Comment by AlenBadel** [comment]
>  I thought there was some other check somewhere that would prevent JIT compilations for platforms that don't support the feature. But I don't recall where it was and if it's based on that flag or not.

There is such a check. 
If J9_EXTENDED_RUNTIME_JIT_INLINE_WATCHES is set then it will trigger this assert on any unsupported arch.
https://github.com/eclipse/openj9/blob/master/runtime/compiler/control/J9Options.cpp#L2531

@gacholio I think it's fair to open issues to track any post implementation work.
Unless we want to broaden the scope of https://github.com/eclipse/openj9/issues/2668 to include this for any JIT cleanup. @cathyzhyi 

The aim of this issue is to address that we failed to enable field watch by default on Z, and P. Simply adding modifying https://github.com/eclipse/openj9/issues/6824#issue-484168252 to include Z, and P will suffice for now.

**Comment by gita-omr** [comment]
> The aim of this issue is to address that we failed to enable field watch by default on Z, and P. Simply adding modifying [#6824 (comment)](https://github.com/eclipse/openj9/issues/6824#issue-484168252) to include Z, and P will suffice for now.

I agree. Sorry for any confusion above. The #if is needed because this  is the flag that VM uses to prevent all JIT compilations on platforms where FW is not supported. This is the one I was looking for.

**Comment by SueChaplain** [comment]
@gita-omr @dchopra001 

If there are any changes required to the draft docs for 0.16.0, please can you raise an issue in the docs repo. 

Current statements in doc here: https://eclipse.github.io/openj9-docs/version0.16/#performance-improvements-for-jvmti-watched-fields-on-power-systems
